File: app/src/main/java/com/softaai/designpatterns/java/structural/decorator/ShapeDecorator.java
package com.softaai.designpatterns.java.structural.decorator;

/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
public class ShapeDecorator implements Shape {
    protected Shape decoratedShape;
    public ShapeDecorator(Shape decoratedShape){
        this.decoratedShape = decoratedShape;
    }
    @Override
    public void draw() {
        decoratedShape.draw();
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/decorator/ShapeDecorator.kt
package com.softaai.designpatterns.kotlin.structural.decorator


/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
open class ShapeDecorator(decoratedShape : Shape) : Shape {
    protected var decoratedShape : Shape

    init {
        this.decoratedShape = decoratedShape
    }
    override fun draw() {
        this.decoratedShape.draw()
    }
}